File: clinic-admin/src/api/clinic.ts
import { get, post } from '@/utils/request'

export interface ClinicInfo {
  id: number
  clinicName: string
  clinicCode: string
  address: string
  province: string
  city: string
  district: string
  contactPerson: string
  contactPhone: string
  logoUrl: string
  licenseNo: string
  licenseUrl: string
  businessStatus: number
  serviceStart: string
  serviceEnd: string
  description: string
  latitude?: number
  longitude?: number
  createdAt?: string
}

export interface ClinicListParams {
  keyword?: string
  status?: number
  pageNum?: number
  pageSize?: number
}

// 获取诊所列表
export const getClinicList = (params: ClinicListParams) => get('/admin/clinic/list', params)

// 获取所有营业中的诊所
export const getActiveClinicList = () => get('/admin/clinic/activeList')

// 获取诊所详情
export const getClinicDetail = (id: number) => get(`/admin/clinic/${id}`)

// 创建诊所
export const createClinic = (data: Partial<ClinicInfo>) => post('/admin/clinic/create', data)

// 更新诊所
export const updateClinic = (data: Partial<ClinicInfo>) => post('/admin/clinic/update', data)

// 删除诊所
export const deleteClinic = (id: number) => post('/admin/clinic/delete', { id })

// 检查诊所编码是否存在
export const checkClinicCode = (clinicCode: string) => get('/admin/clinic/checkCode', { clinicCode })

// 兼容原有接口
export const updateClinicInfo = (data: Partial<ClinicInfo>) => post('/admin/clinic/update', data)

File: clinic-api/src/main/java/com/clinic/entity/Clinic.java
package com.clinic.entity;

import com.baomidou.mybatisplus.annotation.*;
import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.io.Serializable;
import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Data
@TableName("clinic")
public class Clinic implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.AUTO)
    private Long id;

    private String clinicName;
    private String clinicCode;
    private String address;
    private String province;
    private String city;
    private String district;
    private String contactPerson;
    private String contactPhone;
    private String logoUrl;
    private String licenseNo;
    private String licenseUrl;
    private Integer businessStatus;

    // 数据库是 datetime 类型，但前端只传日期，这里用 LocalDateTime 并设置时间为 00:00:00
    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDateTime serviceStart;

    @DateTimeFormat(pattern = "yyyy-MM-dd")
    @JsonFormat(pattern = "yyyy-MM-dd")
    private LocalDateTime serviceEnd;

    private String description;
    private Double latitude;
    private Double longitude;

    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;

    @TableField(fill = FieldFill.INSERT_UPDATE)
    private LocalDateTime updatedAt;

    @TableLogic
    private Integer deleted;

    // 设置日期时，自动转换为 LocalDateTime（时间为 00:00:00）
    public void setServiceStart(LocalDate date) {
        this.serviceStart = date != null ? date.atStartOfDay() : null;
    }

    public void setServiceEnd(LocalDate date) {
        this.serviceEnd = date != null ? date.atStartOfDay() : null;
    }
}
